src/reanimated2/animation/spring.ts

```diff
@@ -144,13 +293,42 @@ export function withSpring(
       previousAnimation: SpringAnimation
     ): void {
       animation.current = value;
+      animation.startValue = value;
+
+      let newMass = config.mass;
+
+      if (userConfig?.duration) {
+        let x0 = Number(animation.toValue) - value;
+        if (Number(previousAnimation?.startValue)) {
+          // It makes our animation a bit smoother, especially if sb triggered same animation twice
+          x0 = Number(animation.toValue) - Number(previousAnimation.startValue);
```

Comment:
I think the behavior in this case should be that we aim to finish at the time when the previous animation started. Therefore this shouldn't be necessary. We shouldn't need to keep `startValue` as a result. Instead, what we should be doing here is to use the current velocity (not the one from `config` but one from `animation` object) as well as use `startTimestamp` from previous animation in order to calculate the remaining time instead of using `duration` from the config.